sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1525,8 +1525,13 @@ object QueryCompilationErrors {
   }
 
   def invalidLiteralForWindowDurationError(): Throwable = {
-    new AnalysisException("The duration and time inputs to window must be " +
-      "an integer, long or string literal.")
+    new AnalysisException(
+      errorClass = "INVALID_PARAMETER_VALUE",
+      messageParameters = Array(
+        "Duration/Time",
+        "window function",
+        "The duration and time " +
+          "inputs to window must be an integer, long or string literal."))
```

Comment:
One key benefit of error classes is that we can move to localized (translated) error messages. So we should avoid "text" in the token - which won't be translated.
Maybe this text is a hint that this message needs its own error class?

Reply:
Got it, will fix!